ldap/config.go

```diff
@@ -0,0 +1,202 @@
+	// contractors, you could write
+	// (&(objectClass=user)({{.UserAttr}}={{.Username}})(!(employeeType=Contractor)))
+	UserFilter string `json:"userfilter"`
+
+	// UserAttr is the "username" attribute of the entry's DN and is typically
+	// either the cn in ActiveDirectory or uid in openLDAP  (default: cn)
+	UserAttr string `json:"userattr"`
+
+	// Certificate to use verify the identity of the directory service and is a
+	// PEM encoded x509 (optional)
+	Certificate string `json:"certificate"`
+
+	// ClientTLSCert is the client certificate used with the ClientTLSKey to
+	// authenticate the client to the directory service.  It must be PEM encoded
+	// x509 (optional)
+	ClientTLSCert string `json:"client_tls_cert"`
+
+	// ClientTLSKey is the client certificate key used with the ClientTLSCert to
+	// authenticate the client to the directory service.  It must be a PEM
+	// encoded x509 (optional)
+	ClientTLSKey string `json:"client_tls_key"`
+
+	// InsecureTLS will skip the verification of the directory service's
+	// certificate when making a client connection (optional).
+	// Warning: this is insecure
+	InsecureTLS bool `json:"insecure_tls"`
+
+	// StartTLS will issue the StartTLS command after establishing an initial
+	// non-TLS connection (optional)
+	StartTLS bool `json:"starttls"`
+
+	// BindDN is the distinguished name used when the client binds
+	// (authenticates) to a directory service
+	BindDN string `json:"binddn"`
+
+	// BindPassword is the password used with the BindDN when the client binds
+	// (authenticates) to a directory service (optional)
+	BindPassword string `json:"bindpass"`
+
+	// AllowEmptyPasswordBinds: if true it allows binds even if the user's
+	// password is empty (zero length) (optional).
+	AllowEmptyPasswordBinds bool `json:"allow_empty_passwd_bind"`
+
+	// DiscoverDN: if true, it will use an anonymous bind with a search
+	// to discover the bind DN of a user (optional)
+	DiscoverDN bool `json:"discoverdn"`
+
+	// TLSMinVersion version to use. Accepted values are
+	// 'tls10', 'tls11', 'tls12' or 'tls13'. Defaults to 'tls12'
+	TLSMinVersion string `json:"tls_min_version"`
+
+	// TLSMaxVersion version to use. Accepted values are 'tls10', 'tls11',
+	// 'tls12' or 'tls13'. Defaults to 'tls12'
+	TLSMaxVersion string `json:"tls_max_version"`
+
+	// UseTokenGroups: if true, use the Active Directory tokenGroups constructed
+	// attribute of the user to find the group memberships. This will find all
+	// security groups including nested ones.",
+	UseTokenGroups bool `json:"use_token_groups"`
+
+	// RequestTimeout in seconds, for the connection when making requests
+	// against the server before returning back an error.
+	RequestTimeout int `json:"request_timeout"`
+
+	// UsePre111GroupCNBehavior: if true, group searching reverts to the pre
+	// 1.1.1 Vault behavior.
+	// see: https://www.vaultproject.io/docs/upgrading/upgrade-to-1.1.1
+	UsePre111GroupCNBehavior *bool `json:"use_pre111_group_cn_behavior"`
```

Comment:
Suggestion: name it something like `DeprecatedVaultPre111GroupCNBehavior`. Unless it's likely people will actually want the previous behavior outside of niche Vault installs, but in that case it should probably not have "Pre111" in it at all and be a more descriptive discussion of what it does.

Reply:
done.  ty!